pydantic/_internal/_model_construction.py

```diff
@@ -243,7 +243,8 @@ def wrapped_model_post_init(self: BaseModel, context: Any, /) -> None:
         else:
             # These are instance variables, but have been assigned to `NoInitField` to trick the type checker.
             for instance_slot in '__pydantic_fields_set__', '__pydantic_extra__', '__pydantic_private__':
-                del namespace[instance_slot]
+                if instance_slot in namespace:  # Safe check in case the metaclass is used other than in `BaseModel`:
+                    del namespace[instance_slot]
```

Comment:
Surely, `nameapace.pop(instance_slot, None)` is better - only one lookup of the dict.